src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5037,6 +5037,12 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_PatternNullableType" xml:space="preserve">
     <value>It is not legal to use nullable type '{0}' in a pattern; use the underlying type '{1}' instead.</value>
   </data>
+  <data name="ERR_IsNullableType" xml:space="preserve">
+    <value>It is not legal to use nullable type '{0}?' in an is-type expression; use the underlying type '{0}' instead.</value>
```

Comment:
I would add "because `is` never matches `null`" to make it clear why it's illegal. I suspect people trying to do this may not have the correct mental model.

Reply:
That's not the reason.  This is only given for nullable reference types.  The reason is that it is a runtime check, and nullable reference types do not exist at runtime.  I'll clarify.

---
In reply to: [249944779](https://github.com/dotnet/roslyn/pull/32431#discussion_r249944779) [](ancestors = 249944779)